### PR TITLE
fix: push notification flush events

### PR DESCRIPTION
**Pull request by tyllark**
[Github Issue 5122](https://github.com/aws-amplify/amplify-flutter/issues/5122)

*Description of changes:*
1. Moved StoppableTimer to aws_common so it can be reused to flush events for push notifications and analytics.
2. [Re-added logic](https://github.com/aws-amplify/amplify-flutter/commit/f0f572868e852b1da393507b37c07cb4110fc56b) for flushing events every 10 seconds.
3. Updated push notification event source names to match [Pinpoint documentation](https://docs.aws.amazon.com/pinpoint/latest/developerguide/event-streams-data-campaign.html#event-streams-data-campaign-attributes).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_core/lib/src/types/notifications/push/pinpoint_event_source.dart [hunk 1]
```diff
@@ -11,8 +11,8 @@
 /// [Journeys](https://docs.aws.amazon.com/pinpoint/latest/userguide/journeys.html)
 /// {@endtemplate}
 enum PinpointEventSource {
```
Comment (NikaHsn) [inline packages/amplify_core/lib/src/types/notifications/push/pinpoint_event_source.dart:13]: is this type public/exported? if so is this considered a breaking change?

Comment (tyllark) [reply packages/amplify_core/lib/src/types/notifications/push/pinpoint_event_source.dart:14]: I marked the enum as deprecated and created a new private enum with the updated values. thanks for the callout!

File: packages/aws_common/lib/src/util/stoppable_timer.dart [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+
+/// {@template amplify_analytics_pinpoint_dart.stoppable_timer}
+/// A Timer that can be stopped and started again.
+/// {@endtemplate}
+@protected
+class StoppableTimer {
```
Comment (Jordan-Nelson) [inline packages/aws_common/lib/src/util/stoppable_timer.dart:12]: Question: Can StoppableTimer be removed from amplify_analytics_pinpoint?

Comment (tyllark) [reply packages/aws_common/lib/src/util/stoppable_timer.dart:12]: Good catch. I initially moved StoppableTimer into amplify_core, but forgot to remove it after migrating it again to aws_common.

**Review by Jordan-Nelson — APPROVED** [review]

**Review by Equartey — APPROVED** [review]

**Review by NikaHsn — APPROVED** [review]

**Pull request merged**